modules/snooker/src/main/ts/ephox/snooker/api/ResizeBehaviour.ts

```diff
@@ -0,0 +1,144 @@
+import { Arr, Fun } from '@ephox/katamari';
+
+type TableResizer = (delta: number) => void;
+
+export interface ResizeBehaviour {
+  readonly resizeTable: (resizer: TableResizer, delta: number, isLastColumn: boolean) => void;
+  readonly clampTableDelta: (sizes: number[], index: number, delta: number, minCellSize: number, isLastColumn: boolean) => number;
+  readonly calcLeftEdgeDeltas: (input: number[], index: number, nextIndex: number, delta: number, minCellSize: number, relativeSizing: boolean) => number[];
+  readonly calcMiddleDeltas: (input: number[], previousIndex: number, index: number, nextIndex: number, delta: number, minCellSize: number, relativeSizing: boolean) => number[];
+  readonly calcRightEdgeDeltas: (input: number[], previousIndex: number, index: number, delta: number, minCellSize: number, relativeSizing: boolean) => number[];
+}
+
+const zero = (array: number[]) => Arr.map(array, Fun.constant(0));
+
+const surround = (input: number[], startIndex: number, endIndex: number, results: number[], f: (array: number[]) => number[]) =>
+  f(input.slice(0, startIndex)).concat(results).concat(f(input.slice(endIndex)));
+
+// Clamp positive or negative delta so that a column/row cannot be reduced past its min size
+const clampDeltaHelper = (predicate: (delta: number) => boolean) => (sizes: number[], index: number, delta: number, minCellSize: number) => {
+  if (!predicate(delta)) {
+    return delta;
+  }
```

Comment:
Please use an `else` here. It produces clearer code and also minifies better.

Reply:
Fixed